test/functional/tests/failover_standby/test_failover_advanced.py

```diff
@@ -163,14 +519,215 @@ def test_failover_during_io_with_eviction(pyocf_ctx, cache_mode, cls, cleaning_p
+
+    with TestRun.use_dut(primary_node), TestRun.step("Verify cache is > 25% dirty"):
+        dirty = int(primary_node.cache.get_statistics().usage_stats.dirty) / int(capacity)
+        assert dirty > 0.25
+
+    with TestRun.use_dut(primary_node), TestRun.step("Verify cache ocuppancy is 100%"):
+        occupancy = int(primary_node.cache.get_statistics().usage_stats.occupancy) / int(capacity)
+        assert occupancy > 0.99
+
+    with TestRun.use_dut(primary_node), TestRun.step("Switch to WO cache mode without flush"):
+        primary_node.cache.set_cache_mode(CacheMode.WO, flush=False)
+
+    with TestRun.use_dut(primary_node), TestRun.step("Calculate checksum of CAS exported object"):
+        checksum1 = TestRun.executor.run(f"md5sum {core.path}").stdout.split()[0]
+
+    with TestRun.use_dut(primary_node), TestRun.step(f"Set {cache_mode} cache mode without flush"):
+        primary_node.cache.set_cache_mode(cache_mode, flush=False)
+
+    with TestRun.use_dut(primary_node), TestRun.step(
+        f"Switch to {cleaning_policy} cleaning policy"
+    ):
+        primary_node.cache.set_cleaning_policy(cleaning_policy)
+
+        if cleaning_policy == CleaningPolicy.alru:
+            TestRun.LOGGER.info("Configure ALRU to trigger immediately\n")
+            params = FlushParametersAlru(
+                activity_threshold=Time(milliseconds=0),
+                wake_up_time=Time(seconds=0),
+                staleness_time=Time(seconds=1),
+            )
+            primary_node.cache.set_params_alru(params)
+
+    with TestRun.step("Wait 2s for cleaner to kick in"):
+        time.sleep(2)
+
+    with TestRun.use_dut(primary_node):
+        stats_before = primary_node.cache.get_statistics()
+
+    with TestRun.use_dut(primary_node), TestRun.step("Run mult-threaded fio"):
+        start_size = Size(4, Unit.KibiByte).get_value()
+        stop_size = int(cls)
+
+        fio = (
+            Fio()
+            .create_command()
+            .direct(True)
+            .read_write(ReadWrite.randread)
+            .blocksize_range([(start_size, stop_size)])
+            .size(size)
+            .file_name(core.path)
+            .rand_seed(TestRun.random_seed)
+            .num_jobs(16)
+            .size(core_size)
+            .time_based(True)
+            .run_time(timedelta(minutes=60))
+        )
+
+        fio.run_in_background()
+
+    with TestRun.step("Wait 2s for I/O to take effect"):
+        time.sleep(2)
+
+    with TestRun.use_dut(primary_node):
+        stats_after = primary_node.cache.get_statistics()
+
+    with TestRun.use_dut(primary_node), TestRun.step(
+        "Verify cache miss statistic is being incremented"
+    ):
+        read_misses_before = (
+            stats_before.request_stats.read.full_misses
+            + stats_before.request_stats.read.part_misses
+        )
+
+        read_misses_after = (
+            stats_after.request_stats.read.full_misses + stats_after.request_stats.read.part_misses
+        )
```

Comment:
Please refactor cohesively, for example:

    with TestRun.use_dut(primary_node), TestRun.step(
            "Verify cache miss statistic is being incremented"):

        read_misses_before = (stats_before.request_stats.read.full_miss 
                                            + stats_before.request_stats.read.part_misses)

        read_misses_after = (stats_after.request_stats.read.full_misses 
                                         + stats_after.request_stats.read.part_misses)

Reply:
This is also generated by black (see my answer to the next comment).

Reply:
ok!